src/classes/Entity/Entity.ts

```diff
@@ -1157,13 +1165,18 @@ error(`'${attrs[i]}' has a default value and cannot be removed`)
                   REMOVE.push(`${path}`)
                 } else if (input.$add) {
                   ADD.push(`${path} :${value}`)
+
                   values[`:${value}`] = input.$add
                 } else if (input.$append) {
-                  SET.push(`${path} = list_append(${path},:${value})`)
+                  SET.push(`${path} = list_append(${path}, if_not_exists(:${value}, :${ATTRIBUTE_VALUES_LIST_DEFAULT_KEY}))`)
```

Comment:
This isn't going to work. `if_not_exists` requires a path not a value. So it should probably be:

```typescript
SET.push(`${path} = list_append(if_not_exists(${path}, :${ATTRIBUTE_VALUES_LIST_DEFAULT_KEY}), :${value})`)
```
But even with this fixed, we need to default a value of `{}` for each element in the path.

Reply:
It looks like DynamoDB won't support overlapping paths. So with the fix I just put in, `$append` and `$prepend` in a nested map will only work if the _parent path_ exists. For example, you can append to `test.test1.test2` only if a root map attribute of `test` already exists in the item with an object property of `test1`. I don't think there is a way around that.

However, previously, your map would also need to have the final item in the path to already exists, e.g.:

 ```javascript
test: { 
  test1: {
    test2: []
  }
}
```

With this update, at least you can dynamically append/prepend to non-existing child items.